frontend/packages/console-dynamic-plugin-sdk/src/extensions/topology-details.ts

```diff
@@ -106,3 +147,15 @@ export type DetailsResourceAlertContent = {
   variant?: 'success' | 'danger' | 'warning' | 'info' | 'default';
   actionLinks?: React.ReactNode;
 };
+
+export type AdapterDataType<D = {}> = {
+  resource: K8sResourceCommon;
+  provider?: (resource: K8sResourceCommon) => D;
```

Comment:
provider is optional because some adpater might not need to provide a hook. For example network adapter doesn't need to provide a hook just the resource.

Reply:
Shouldn't you use a specific data object per adapter type? Each adapter return value is specific. So for the case where you must return a provider, now the code must also do a null check which is incorrect.